zipline/errors.py

```diff
@@ -858,3 +858,14 @@ class IncompatibleTerms(ZiplineError):
         "{term_1} and {term_2} must have the same mask in order to compute "
         "correlations and regressions asset-wise."
     )
+
+
+class BoundColumnInvalidCompare(ZiplineError):
+    """
+    Raised when trying to use BoundColumn with a comparison operator.
+    Frequently occurs when users forget to use '.latest'.
+    """
+    msg = (
+        "Can't compare BoundColumn with {other}.  Did you mean to append "
```

Comment:
Two notes on this:

1. I would probably render `type(other)` rather than `other` here. Objects, especially complex pipeline expressions, can have very large string representations, which tend to be less helpful in errors like this.
2. This message should probably include the `__str__` of the column being compared. Saying "You can't compare `EquityPricing.close` to an `int` is more likely to be helpful for a user than "You can't compare BoundColumns to ints." The existing error we raise on invalid inputs to `Pipeline` does a good job of this IMO.

Reply:
Sounds good.  Regarding 1., right now I am passing `type(other)` as the kwarg value to `BoundColumnInvalidCompare` when I raise the exception.  Is there a better way to do that?

Agreed on 2., fixing now.

Reply:
> Is there a better way to do that?

You could override the `__init__` of your error to forward `type(other)` rather than requiring it at each call-site? 

In general, I'm not really sure the implicit formatting behavior of the `ZiplineError` hierarchy here is worth the confusion it causes, but given all the stuff that's using it I don't think we should change it.

Reply:
Oh nvm, I found {x.type} in python string formatting

Reply:
Ended up going with {other.__class__} so I don't have to call type(other) at each call-site